### PR TITLE
Jason itinerary view

### PR DESCRIPTION
**Pull request by Topia99**
Itinerary View
1. display a each day planning in a trip itinerary
2. display a list of items in each day planning
3. allow add and edit day item

Trip list
1. display a list of trip itinerary
2. allow add new trip itinerary

### CODE REVIEW

File: Travel-Planner/Travel-Planner/Screens/Itinerary/TripView.swift [hunk 1]
```diff
@@ -0,0 +1,59 @@
+//
+//  ContentView.swift
+//  Travel-Planner
+//
+//  Created by Jason Zeng on 9/26/24.
+//
+
+import SwiftUI
+import Foundation
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Screens/Itinerary/TripView.swift]: no need for Foundation. Remove it.

File: Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlan/DayPlanTitleView.swift [hunk 2]
```diff
@@ -0,0 +1,43 @@
+//
+//  DateplanTitle.swift
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlan/DayPlanTitleView.swift]: Not used. Remove

File: Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlanItem/TripItemDetail.swift [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+    var body: some View {
+        HStack {
+            Image(systemName: "location")
+            TextField("Location", text: $location)
+        }
+    }
+}
+
+//struct TimeInputView: View {
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlanItem/TripItemDetail.swift]: removed it

File: Travel-Planner/Travel-Planner/Views/ItineraryView/HScrollDatesView.swift [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+                }
+            }
+            .padding(.horizontal)
+        }
+        .frame(height: 50)
+    }
+}
+
+//#Preview {
+//    HScrollDatesView(dayPlans: MockData.sampleItinerary.dayPlans, selectedDate: .constant(Date.now))
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Views/ItineraryView/HScrollDatesView.swift]: removed preview

**Review by Topia99 — COMMENTED** [review]
need small fix

File: Travel-Planner/Travel-Planner/Screens/Itinerary/TripView.swift (hunk 1 above)
Comment (Topia99) [reply Travel-Planner/Travel-Planner/Screens/Itinerary/TripView.swift]: fix

File: Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlan/DayPlanTitleView.swift (hunk 2 above)
Comment (Topia99) [reply Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlan/DayPlanTitleView.swift]: removed

File: Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlanItem/TripItemDetail.swift (hunk 3 above)
Comment (Topia99) [reply Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlanItem/TripItemDetail.swift]: removed

File: Travel-Planner/Travel-Planner/Views/ItineraryView/HScrollDatesView.swift (hunk 4 above)
Comment (Topia99) [reply Travel-Planner/Travel-Planner/Views/ItineraryView/HScrollDatesView.swift]: removed

**Review by Topia99 — COMMENTED** [review]
Reviewed by author

**Review by SamYang10086 — APPROVED** [review]
approved

**Pull request merged**